django/contrib/sessions/backends/base.py

```diff
@@ -147,24 +156,60 @@ def is_empty(self):
         except AttributeError:
             return True
 
+    def _get_session_key_part(self, session_key):
+        "Return a string of the session_key key."
+        try:
+            algorithm, key = session_key.split(SESSION_KEY_DELIMITER)
+            if algorithm == SESSION_HASHING_ALGORITHM:
+                return key
+            return None
+        except ValueError:
+            return None
+
+    def _is_hashed(self, session_key):
+        "Return True when the session_key is hashed in the backend."
+        try:
+            return bool(self._get_session_key_part(session_key))
+        except IndexError:
+            return False
+
+    def get_session_key_hash(self, session_key):
+        """
+        Return a hash of the session key to be stored in the db. We want a fast
+        hashing function to minimize the impact on performance. We don't have to
+        worry about speed and salting since the session_key is already high entropy.
+        """
+        if session_key is not None:
+            if self._is_hashed(session_key) or settings.SESSION_REQUIRE_KEY_HASH:
+                key = self._get_session_key_part(session_key)
+                if key:
+                    return self._session_hashing_algorithm(key.encode('ascii')).hexdigest()
+            return session_key
```

Comment:
This method is called `get_session_key_hash` but it returns the key without hashing here.

I suggest to rename the method to `maybe_hash_session_key` or something along these lines.